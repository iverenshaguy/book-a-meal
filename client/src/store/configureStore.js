import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rootReducer, { history } from '../reducers/rootReducer';

const enhancers = [];
const middlewares = [thunk, routerMiddleware(history)];


if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);

  const { devToolsExtension } = window;

  if (typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension());
  }
}

const composedEnhancers = compose(applyMiddleware(...middlewares), ...enhancers);

export default createStore(rootReducer, composedEnhancers);
