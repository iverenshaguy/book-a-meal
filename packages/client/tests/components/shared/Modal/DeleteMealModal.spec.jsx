import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { initialState, mealsObj } from '../../../setup/mockData';
import DeleteMealModalComponent from '../../../../src/components/shared/Modal/DeleteMealModal';
import DeleteMealModalContainer from '../../../../src/containers/shared/Modal/DeleteMealModal';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({
  ...initialState,
  meals: { ...initialState.meals, items: mealsObj.meals }
});
const props = {
  editMeal: jest.fn(),
  deleteMeal: jest.fn(),
  toggleModal: jest.fn(),
  deleting: false,
  meal: mealsObj.meals[1]
};

describe('DeleteMealModal', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<DeleteMealModalComponent {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.delete-meal').length).toEqual(1);
    expect(wrapper.find('MiniPreloader').length).toBeFalsy();
  });

  it('should render MiniPreloader when deleting', () => {
    const wrapper = shallow(<DeleteMealModalComponent {...props} deleting />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('MiniPreloader').length).toEqual(1);
    expect(wrapper.find('.delete-meal').length).toBeFalsy();
  });

  it('should render connected component correctly', () => {
    const comp = (<Provider store={store}><DeleteMealModalContainer {...props} /></Provider>);
    const wrapper = mount(comp);

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should call toggleModal when delete meal is not confirmed', () => {
    const toggleModalMock = jest.fn();
    const comp = (
      <Provider store={store}>
        <DeleteMealModalComponent {...props} toggleModal={toggleModalMock} />
      </Provider>
    );
    const wrapper = mount(comp);

    wrapper.find('#confirm-delete-no').simulate('click');

    expect(toggleModalMock).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('should call handleDeleteClick when delete meal is confirmed', () => {
    const deleteMealMock = jest.fn();
    const comp = (
      <Provider store={store}>
        <DeleteMealModalComponent {...props} deleteMeal={deleteMealMock} />
      </Provider>
    );
    const wrapper = mount(comp);

    wrapper.find('#confirm-delete-yes').simulate('click');

    expect(deleteMealMock).toHaveBeenCalled();
    wrapper.unmount();
  });
});
