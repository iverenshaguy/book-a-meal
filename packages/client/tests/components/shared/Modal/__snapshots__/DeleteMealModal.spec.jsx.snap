// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMealModal should render MiniPreloader when deleting 1`] = `
<Fragment>
  <div
    className="text-center"
  >
    <MiniPreloader />
  </div>
</Fragment>
`;

exports[`DeleteMealModal should render connected component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(DeleteMealModal)
    deleteMeal={[MockFunction]}
    deleting={false}
    editMeal={[MockFunction]}
    meal={
      Object {
        "description": "Sharwama contains no animal products, perfect for dieters",
        "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
        "price": "1200.00",
        "title": "Vegetable Sharwama and Guava Smoothie",
        "vegetarian": true,
      }
    }
    toggleModal={[MockFunction]}
  >
    <DeleteMealModal
      deleteMeal={[Function]}
      deleting={false}
      editMeal={[MockFunction]}
      meal={
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        }
      }
      toggleModal={[MockFunction]}
    >
      <div
        className="delete-meal"
      >
        <p>
          Are You Sure?
        </p>
        <div
          className="confirm-delete-btns control-btns"
        >
          <button
            className="btn btn-sec"
            id="confirm-delete-no"
            onClick={[Function]}
            type="button"
          >
            No
          </button>
          <button
            className="btn btn-sec-danger"
            id="confirm-delete-yes"
            onClick={[Function]}
            type="button"
          >
            Yes
          </button>
        </div>
      </div>
    </DeleteMealModal>
  </Connect(DeleteMealModal)>
</Provider>
`;

exports[`DeleteMealModal should render correctly 1`] = `
<Fragment>
  <div
    className="delete-meal"
  >
    <p>
      Are You Sure?
    </p>
    <div
      className="confirm-delete-btns control-btns"
    >
      <button
        className="btn btn-sec"
        id="confirm-delete-no"
        onClick={[Function]}
        type="button"
      >
        No
      </button>
      <button
        className="btn btn-sec-danger"
        id="confirm-delete-yes"
        onClick={[Function]}
        type="button"
      >
        Yes
      </button>
    </div>
  </div>
</Fragment>
`;
